src/os/windows/library_loader_win.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file library_loader_win.cpp Implementation of the LibraryLoader for Windows */
+
+#include "../../stdafx.h"
+
+#include <windows.h>
+
+#include "../../library_loader.h"
+
+#include "../../safeguards.h"
+
+static std::string GetLoadError()
+{
+	auto error = GetLastError();
+	std::string message;
+	char buffer[512];
+	if (FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr, error,
+		MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), buffer, sizeof(buffer), nullptr) == 0) {
+		message.assign(fmt::format("Unknown error {}", error));
+	} else {
+		message.assign(buffer);
+	}
+	return message;
```

Comment:
```suggestion
	char buffer[512];
	if (FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr, error,
		MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), buffer, sizeof(buffer), nullptr) == 0) {
		return fmt::format("Unknown error {}", error);
	}
	return buffer;
```
This saves at least the copy that is happening due to the `assign`. And if the compiler can do copy elision/return value optimization, it will write directly in the structure of the caller of this function.